# Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Position Controllers --------------------------------------
  rear_left_steering_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: rear_left_steering_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  rear_left_wheel_joint_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: rear_left_wheel_joint
    pid: {p: 100, i: 20, d: 0.0, i_clamp: 100}
  front_right_steering_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: front_right_steering_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  front_left_steering_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: front_left_steering_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  rear_right_steering_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: rear_right_steering_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  rear_right_wheel_joint_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: rear_right_wheel_joint
    pid: {p: 100, i: 20, d: 0.0, i_clamp: 100}
  front_right_wheel_joint_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: front_right_wheel_joint
    pid: {p: 100, i: 20, d: 0.0, i_clamp: 100}
  front_left_wheel_joint_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: front_left_wheel_joint
    pid: {p: 100, i: 20, d: 0.0 ,i_clamp: 100}
  arm_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: arm_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  gripper_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: gripper_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  slider_position_controller:
    type: effort_controllers/JointPositionController
    joint: slider
    pid: {p: 100.0, i: 0.01, d: 10.0}
